<a href="https://colab.research.google.com/github/MInesGomes/DataScienceObesityLevels/blob/main/DSGroup2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**1. Preprocessing and Exploratory Data Analysis (EDA)**

• Handling missing values and inconsistencies

• Encoding categorical variables

• Feature scaling and transformation

• Identifying outliers

• Exploring relationships between variables through visualizations

In [ ]:
import pandas as pd
import numpy as np

df = pd.read_csv('/content/ObesityDataSet_raw_and_data_sinthetic.csv')

print("\nDataset info:")
print(df.info())
print("\nFirst 5 rows of the dataset:")
display(df.head())


Dataset info:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2111 entries, 0 to 2110
Data columns (total 17 columns):
 #   Column                          Non-Null Count  Dtype  
---  ------                          --------------  -----  
 0   Gender                          2111 non-null   object 
 1   Age                             2111 non-null   float64
 2   Height                          2111 non-null   float64
 3   Weight                          2111 non-null   float64
 4   family_history_with_overweight  2111 non-null   object 
 5   FAVC                            2111 non-null   object 
 6   FCVC                            2111 non-null   float64
 7   NCP                             2111 non-null   float64
 8   CAEC                            2111 non-null   object 
 9   SMOKE                           2111 non-null   object 
 10  CH2O                            2111 non-null   float64
 11  SCC                             2111 non-null   object 
 12  FAF                

,Gender,Age,Height,Weight,family_history_with_overweight,FAVC,FCVC,NCP,CAEC,SMOKE,CH2O,SCC,FAF,TUE,CALC,MTRANS,NObeyesdad
0,Female,21.0,1.62,64.0,yes,no,2.0,3.0,Sometimes,no,2.0,no,0.0,1.0,no,Public_Transportation,Normal_Weight
1,Female,21.0,1.52,56.0,yes,no,3.0,3.0,Sometimes,yes,3.0,yes,3.0,0.0,Sometimes,Public_Transportation,Normal_Weight
2,Male,23.0,1.80,77.0,yes,no,2.0,3.0,Sometimes,no,2.0,no,2.0,1.0,Frequently,Public_Transportation,Normal_Weight
3,Male,27.0,1.80,87.0,no,no,3.0,3.0,Sometimes,no,2.0,no,2.0,0.0,Frequently,Walking,Overweight_Level_I
4,Male,22.0,1.78,89.8,no,no,2.0,1.0,Sometimes,no,2.0,no,0.0,0.0,Sometimes,Public_Transportation,Overweight_Level_II


 The dataset contains 2111 entries and 17 columns.

 There are no missing values in any of the columns, as indicated by the 'Non-Null Count'.

 The columns include both numerical (e.g., Age, Height, Weight) and categorical (e.g., Gender, family_history_with_overweight, NObeyesdad) data types.

In [ ]:
#family_history_with_overweight and other boolean columns-> could be replaced by 0 or 1 ?


# MTRANS will be useful to the question: Do transportation habits correlate with healthier lifestyles?

**2. Additional Insights**

*  Do physical activity levels significantly influence obesity classification?
*  Are eating habits stronger predictors of obesity than demographic factors?
*  Is there a relationship between technology usage (screen time) and obesity levels?
*  Do transportation habits correlate with healthier lifestyles?
*  Can obesity patterns vary across age groups or genders?

In [ ]:
#code here

**3. Predictive Modeling**

* Logistic Regression
* Decision Trees
* Random Forest
* Support Vector Machines
* K-Nearest Neighbors
* Gradient Boosting methods

**Students should:**

* Select relevant features
* Train and evaluate multiple models
* Compare model performance
* Apply hyperparameter tuning
* Interpreting the most important predictors


**Performance should be evaluated using appropriate metrics such as:**
* Accuracy
* Precision
* Recall

In [ ]:
#code here

**NObeyesdad** represents the obesity classification of each individual.

The categories include:

* Insufficient Weight
* Normal Weight
* Overweight Level I
* Overweight Level II
* Obesity Type I
* Obesity Type II
* Obesity Type III

You will use this variable as the target variable for predictive modeling.